apps/builder/app/builder/features/settings-panel/props-section/animation/animation-keyframes.tsx

```diff
@@ -157,25 +170,49 @@ const Keyframe = ({
         </Tooltip>
       </Grid>
       <Grid>
-        <CodeEditor
-          lang="css-properties"
-          size="keyframe"
-          value={cssProperties}
-          onChange={() => {
-            /* do nothing */
+        <CssEditor
+          showSearch={false}
+          propertiesPosition="top"
+          virtualize={false}
+          styleMap={styleMap}
+          apiRef={apiRef}
+          onAddDeclarations={(addedStyleMap) => {
+            const styles = { ...value.styles };
+            for (const [property, value] of addedStyleMap) {
+              styles[property] = value;
+            }
+            onChange({ ...value, styles });
+          }}
+          onDeleteProperty={(property, options = {}) => {
+            if (options.isEphemeral === true) {
+              return;
+            }
+            const styles = { ...value.styles };
+            delete styles[property];
+            onChange({ ...value, styles });
           }}
-          onChangeComplete={(cssText) => {
-            const parsedStyles = parseCss(`selector{${cssText}}`);
-            onChange({
-              ...value,
-              styles: parsedStyles.reduce(
-                (r, { property, value }) => ({ ...r, [property]: value }),
-                {}
-              ),
-            });
+          onSetProperty={(property) => {
+            return (newValue) => {
```

Comment:
I would remove this carrying from CssEditor api

Reply:
I just wanted to keep the exact same interface as we have with setProperty and void having different variations of it.

Reply:
I would remove that carrying everywhere 😄